src/helper/rsa.rs

```diff
@@ -0,0 +1,205 @@
+use std::{fmt, marker::PhantomData};
+
+use chrono::{DateTime, Utc};
+use rand::{CryptoRng, Rng};
+use rsa::{
+    pkcs1v15::{Signature, VerifyingKey},
+    traits::PublicKeyParts,
+    RsaPublicKey,
+};
+use sha2::Digest;
+use signature::{hazmat::PrehashSigner, Keypair, SignatureEncoding};
+
+use crate::{
+    bail,
+    crypto::{hash::HashAlgorithm, public_key::PublicKeyAlgorithm},
+    errors::Result,
+    packet::PublicKey,
+    types::{
+        EskType, Fingerprint, KeyId, KeyVersion, Mpi, PkeskBytes, PublicKeyTrait, PublicParams,
+        SecretKeyTrait, SignatureBytes, Version,
+    },
+};
+
+use super::{PgpHash, PgpPublicKey};
+
+impl PgpPublicKey for RsaPublicKey {
+    const PGP_ALGORITHM: PublicKeyAlgorithm = PublicKeyAlgorithm::RSA;
+
+    fn pgp_parameters(&self) -> PublicParams {
+        PublicParams::RSA {
+            n: self.n().into(),
+            e: self.e().into(),
+        }
+    }
+}
+
+/// [`signature::Signer`] backed signer for PGP.
+pub struct RsaSigner<T, D> {
+    inner: T,
+    public_key: PublicKey,
+    _digest: PhantomData<D>,
+}
+
+impl<D, T> fmt::Debug for RsaSigner<T, D> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("RsaSigner").finish()
+    }
+}
+
+impl<T, D> RsaSigner<T, D>
+where
+    D: Digest,
+    T: Keypair<VerifyingKey = VerifyingKey<D>>,
+{
+    /// Create a new signer with a given public key
+    pub fn new(inner: T, created_at: DateTime<Utc>) -> Result<Self> {
+        let public_key = PublicKey::new(
+            Version::New,
+            KeyVersion::V4,
```

Comment:
Too bad that this supports only the V4 version. I'll ping our V6 evangelist here: @hko-s ;)

Reply:
You know a whole lot more about PGP than I do.